site/tests/integration/api-keys.test.ts

```diff
@@ -103,4 +104,41 @@ test("API key page should generate and show a key", async ({
 
   await expect(page.locator(`text=${publicKeyId2}`)).toBeVisible();
   await expect(page.locator(`text=${publicKeyId}`)).not.toBeVisible();
+
+  // @todo Consider moving to a separate test
+  // (requires another run of API key generation logic, perhaps without UI)
+
+  const response1 = await request.get("/api/blocks");
+  expect(response1.status()).toBe(401);
+  expect(await response1.json()).toEqual({
+    errors: [
+      { msg: "A valid API key must be provided in a 'x-api-key' header." },
+    ],
+  });
+
+  const response2 = await request.get("/api/blocks", {
+    headers: { "x-api-key": apiKeyValue },
+  });
+  expect(response2.status()).toBe(401);
+  expect(await response2.json()).toEqual({
+    errors: [{ msg: "API key has been revoked." }],
+  });
+
+  const response3 = await request.get("/api/blocks", {
+    headers: { "x-api-key": apiKeyValue.replace(/\d{4}$/, "0000") },
+  });
+  expect(response3.status()).toBe(401);
+  expect(await response3.json()).toEqual({
+    errors: [{ msg: "API key has been revoked." }],
```

Comment:
Seems this should be "Invalid API key" based on CI?

```suggestion
    errors: [{ msg: "Invalid API key." }],
```

Reply:
Thanks for spotting this! Addressed in https://github.com/blockprotocol/blockprotocol/pull/458/commits/08aa61edace78f2d7934689409456e57c2cbc4a7